pkg/registries/adapters/adapter.go

```diff
@@ -91,8 +93,14 @@ func imageToSpec(log *logging.Logger, req *http.Request, image string) (*apb.Spe
 		return nil, nil
 	}
 
-	err = json.NewDecoder(resp.Body).Decode(&hist)
+	// resp.Body is an io.Reader, which are a one time use.  Save the
+	// contents to a byte[] for debugging, then remake the io.Reader for the
+	// JSON decoder.
+	debug, _ := ioutil.ReadAll(resp.Body)
+	r := bytes.NewReader(debug)
+	err = json.NewDecoder(r).Decode(&hist)
 	if err != nil {
+		log.Errorf("Image '%s' may not exist in registry: %s", image, debug)
```

Comment:
I think using the registry name here, rather then the response body, might make more sense and then you can remove the copy of the byte array?

Reply:
This error check really is catching two error scenarios: A json decoding error and a request error.  I'm going to change this to two different checks by looking for a bad http status code.

I thought about that adding in the registry, but it's embedded in the image variable: ```http://<registry>/<image>/<tag>```.